botblocker.go

```
@@ -47,34 +51,49 @@ func (b *BotBlocker) update() error {
 }
 
-func (b *BotBlocker) updateIps() error {
-	prefixBlockList := make([]netip.Prefix, 0)
+// fetchPrefixes fetches and parses the CIDR lists at the given URLs.
+func fetchPrefixes(urls []string) ([]netip.Prefix, error) {
+	prefixList := make([]netip.Prefix, 0)
 
-	log.Info("Updating CIDR blocklist")
-	for _, url := range b.IpBlocklistUrls {
+	for _, url := range urls {
 		resp, err := http.Get(url)
```

golangci-lint (2.13.2)

[error] net/http.Get must not be called. use net/http.NewRequestWithContext and (*net/http.Client).Do(*http.Request)

(noctx)